src/client/screens/ImportProjectScreen.tsx

```diff
@@ -156,37 +181,51 @@ const ImportProjectScreen = ({ regionConfigs }: StateProps) => {
   const [importResource, setImportResource] = useState<ImportResource>({
     data: null
   });
+  const [rowFlags, setRowFlags] = useState<readonly ImportRowFlag[] | undefined>(undefined);
   const [createProjectResource, setCreateProjectResource] = useState<
     WriteResource<ConfigurableForm, IProject>
   >({
     data: blankForm
   });
+  const [fileError, setFileError] = useState<string | undefined>();
   const regionConfig = importResource.data;
   const formData = createProjectResource.data;
 
   const setFile = useCallback(
     (file: Blob) => {
       async function setConfigFromFile() {
         if (file && "resource" in regionConfigs) {
+          // Check file size (must be less than MaxUploadFileSize)
+          file.size > MaxUploadFileSize && setFileError("File must be less than 25mb");
+          // @ts-ignore I don't think that there is any way this can happen, but it was mentioned in the issue
+          file.type !== "text/csv" && setFileError("File must be a .csv");
```

Comment:
By continuing on afterwards, these more specific errors are getting swallowed by the generic "File must have at least one record".

```suggestion
          if (file.size > MaxUploadFileSize) {
            setFileError("File must be less than 25mb");
            return;
          }
```

Separately, not sure why you had the `// @ts-ignore` there - I was able to remove it w/ no compilation issues.

As for how you could have a `file.type` that isn't "text/csv"
 - In Chrome at least I can switch the pre-selected `.csv` file extension to "Any files"
 
_However_ from my testing relying on `file.type` isn't fool-proof - the CSV file I downloaded from Dave's Redistricting App had an empty string for it's `file.type` when I uploaded it to our site (it seems to be a property of the file?? not sure on this). I think we could still use `file.type` if it is present, but we'd maybe want to fall back to checking if `file.name` ends with `.csv` otherwise.